pkg/gameserverallocations/find.go

```diff
@@ -77,6 +78,19 @@ func findGameServerForAllocation(gsa *allocationv1.GameServerAllocation, list []
 		for j, sel := range gsa.Spec.Selectors {
 			if selectors[j] == nil && sel.Matches(gs) {
 				selectors[j] = &result{gs: gs, index: i}
+				// Once we have found a matching game server for allocation, perfom any Counter or List actions.
```

Comment:
I think we should do this before assign the allocation result to the result list (line 80). I think if the selected GameServer cannot meet the required counter/list action, we should continue the loop and move on to the next GameServer in the list for allocation. Otherwise, there will be scenario that customers do have GameServers that is capable for the required counter/list action but our logic never reaches it and the allocation never succeeds.

Reply:
There is filtering of the GameServers in sel.Matches(gs) on line 79, although that does assume that the caller properly filtered the GameServers. @markmandel Is this the behavior we want, or should we attempt to combine the filtering and allocation actions into one action?